### PR TITLE
ARROW-6139: [Documentation][R] Build R docs (pkgdown) site and add to arrow-site

### PR DESCRIPTION
**Pull request by nealrichardson**
This renders the site that was configured in https://issues.apache.org/jira/browse/ARROW-5452, using the 0.14.1 R CRAN (so-called) "release" branch. Currently there are no links to and from the main arrow.apache.org/docs/ site, but I'd like to take that up in a followup issue because there's more cleanup to do there than just adding a link.

This patch also includes the last two patches to the website source in apache/arrow, which have not yet been published, and it has some relative URL fixes from my fork of the site source.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Full site preview at https://enpiar.com/arrow-site/ and R docs at https://enpiar.com/arrow-site/docs/r/

**Comment by fsaintjacques** [comment]
I don't see the R documentation in the drop-down menu of documentations.

**Comment by nealrichardson** [comment]
> Currently there are no links to and from the main arrow.apache.org/docs/ site, but I'd like to take that up in a followup issue because there's more cleanup to do there than just adding a link.

File: assets/.sprockets-manifest-af5b3e33562477ba4f207e3ea1e798c0.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (kou) [inline assets/.sprockets-manifest-af5b3e33562477ba4f207e3ea1e798c0.json]: Should we remove this?

Comment (nealrichardson) [reply assets/.sprockets-manifest-af5b3e33562477ba4f207e3ea1e798c0.json]: This is autogenerated by jekyll. There's some sorcery in the Jekyll/bootstrap asset rendering that I intend to clean up once https://lists.apache.org/thread.html/73e8a7d2ee667e83371e8bb99c13338144c8af5444f55918994713be@%3Cdev.arrow.apache.org%3E is resolved.

**Comment by nealrichardson** [comment]
This has been revised to build the CRAN published 0.14.1.1 package, and it removes all changes outside of the new `docs/r/` directory.

**Comment by wesm** [comment]
Thanks!

**Pull request merged**